rust-peer/src/main.rs

```diff
@@ -364,3 +428,91 @@ async fn read_or_create_identity(path: &Path) -> Result<identity::Keypair> {
 
     Ok(identity)
 }
+
+// Simple file exchange protocol
+
+#[derive(Debug, Clone)]
+struct FileExchangeProtocol();
+#[derive(Clone)]
+struct FileExchangeCodec();
+#[derive(Debug, Clone, PartialEq, Eq)]
+struct FileRequest(String);
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct FileResponse(Vec<u8>);
+
+impl ProtocolName for FileExchangeProtocol {
+    fn protocol_name(&self) -> &[u8] {
+        "/universal-connectivity-file/1".as_bytes()
+    }
+}
+
+#[async_trait]
+impl request_response::Codec for FileExchangeCodec {
```

Comment:
In the PR description you said you used streaming but this isn't using streaming. That is okay, I just wanted to make sure there isn't a misunderstanding here. This will read the entire file into memory.

Reply:
I am not sure how good the CBOR libraries are for go and JS so depending on that, we may actually want to stick to this codec here.

Reply:
Thank you for your reviews! You’re right. I used streaming for JS (as described [here](https://github.com/libp2p/js-libp2p/tree/master/examples/protocol-and-stream-muxing)), but not for Rust. I’ll change the PR description to avoid misunderstandings. 

> I am not sure how good the CBOR libraries are for go and JS so depending on that, we may actually want to stick to this codec here.

I agree. I didn’t investigate about CBOR in Go and JS yet. After completing this PR using this codec, I’m probably going to try to adopt CBOR and JSON to this file transfer feature, which has been recently added to rust-libp2p by you. For that, I’ll try to see how good the CBOR supports are Go and JS.